docs: Update private-packages.md nuget section (#16942)
renovatebot/renovate: docs/usage/getting-started/private-packages.md

### nuget

For each known NuGet registry, Renovate searches for `hostRules` with `hostType=nuget` and matching host.
For those found, a command like the following is run: `dotnet nuget add source ${registryInfo.feedUrl} --configfile ${nugetConfigFile} --username ${username} --password ${password} --store-password-in-clear-text`.

```js
hostRules: [
  {
    matchHost: 'https://pkgs.dev.azure.com/<org>/',
    hostType: 'nuget',

>     username: 'user',

Reviewer: ```suggestion
    username: 'user', // doesn't matter for azure
```

We should note that this is irrelevant fot azure

### nuget

For each known NuGet registry, Renovate searches for `hostRules` with `hostType=nuget` and matching host.

> For those found, a command similar to the following is
> run: `dotnet nuget add source ${registryInfo.feedUrl} --configfile ${nugetConfigFile} --username ${username} --password ${password} --store-password-in-clear-text`

Reviewer 2: ```suggestion
For those found, a command like the following is run: `dotnet nuget add source ${registryInfo.feedUrl} --configfile ${nugetConfigFile} --username ${username} --password ${password} --store-password-in-clear-text`.
```

- One sentence per line
- End sentence with `.`
- Use simple words, replace `a command similar to the following` with `a command like the following`

```js
hostRules: [
  {
    matchHost: 'https://pkgs.dev.azure.com/<org>/',
    hostType: 'nuget',
    username: 'user',

>     password: '<Encrypted PAT Token>',

Reviewer: what means `Encrypted` here? If it's renovate encryption, it needs to  be
```suggestion
    encryped: { password: '<Encrypted PAT Token>' },
```

Author: @viceice i copied from the discussion https://github.com/renovatebot/renovate/discussions/10998
and from other examples

Reviewer: ```suggestion
    password: '<PAT>',
```

https://github.com/renovatebot/renovate/discussions/10998#discussioncomment-1086730

For each known NuGet registry, Renovate searches for `hostRules` with `hostType=nuget` and matching host.
For those found, a command similar to the following is run: `dotnet nuget add source ${registryInfo.feedUrl} --configfile ${nugetConfigFile} --username ${username} --password ${password} --store-password-in-clear-text`

> password should be encoded to base64

Reviewer 2: ```suggestion
The `password` must be base64 encoded.
```

I think we mean _must_ instead of _should_ here. Or do we mean "We recommend you use base64 encoding, because other methods are wrong/dangerous?"

Are users allowed to use another encoding, or will that break things?

Author: according to rhys's comments its a must , but im not sure since there's no clear documentation, and that's what im trying to clarify here

Author: https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file#packagesourcecredentials
 
 in renovate we recommend encryption, but if you look at the link you will see
` If a non-encrypted password is passed for password the error message ["The parameter is incorrect" will occur](https://github.com/NuGet/Home/issues/3245).`

 its not clear if its base 64 or not, but it should be encrypted i guess,
 apparently there's another parameter called: ClearTextPassword
 for storing the actual password not encrypted.
 i'm not 100% sure how this is implemented so might need help here from @rarkins @viceice 
 
 if they cant answer this then i will have to investigate further

Author: i think we can remove this message for now, it's not necessary as in my example i wrote : Encrypted PAT token

Reviewer: it needs to be plain text password or an API key, depends on the registry.

renovate encryption can be used, but never use base64 encoding.
that can maybe required by some custom registry, but not some i know / use.

Author: @viceice if you have experience here, would you please add some info to the documentation to make it clearer for users?
i've noticed we lack private packages handling documentation for most managers and it's costing us a lot of discussions and time

Reviewer 3: I think there's maybe a mixup here regarding the npm password, where it's base64 encoded in `.npmrc` but we need it non-encoded in `hostRules`. I have no opinion on nuget passwords

Reviewer: I'm simply use this host rule for my `proget` server and it works for `npm` and `nuget` datasources, `SERVICE_USER_PSW` is the plain text password passed via environment.

```js
  {
      matchHost: 'https://proget.domain.test/',
      username: 'service-user',
      password: process.env.SERVICE_USER_PSW,
    },
```

gitea nuget sample, they use plain password or PAT.
- https://docs.gitea.io/en-us/packages/nuget/#configuring-the-package-registry

> I think there's maybe a mixup here regarding the npm password, where it's base64 encoded in `.npmrc` but we need it non-encoded in `hostRules`. I have no opinion on nuget passwords

Yes, seemd to wrongly mixed up from npmrc

Author: in the discussion https://github.com/renovatebot/renovate/discussions/10998 they say they use PAT,
and it has 3 thumps up from other ppl too

> ```js

Reviewer 2: ```suggestion
```json
```

JS: JavaScript
JSON: Javascript Object Notation <-- The one we want to use here

Author: i actually wanted to be consistent with other managers in the same page which also use js instead of json

hostRules: [
    {
      matchHost: "https://pkgs.dev.azure.com/<org>/",
      hostType: "nuget",
      username: "user",
      password: "<Encrypted PAT Token>"
    }

>   ]

Reviewer 2: Make sure the code example is valid JSON, right now its wrong.
